compass/Coordinator.java

```diff
@@ -240,33 +307,38 @@ private void start() throws ArgumentException, InterruptedException {
         }
       }
 
-      latestMilestone++;
+      // If all the above checks pass we are ready to issue a new milestone
+      state.latestMilestoneIndex++;
 
-      log.info("Issuing milestone: " + latestMilestone);
+      log.info("Issuing milestone: " + state.latestMilestoneIndex);
       log.info("Trunk: " + trunk + " Branch: " + branch);
-      List<Transaction> txs = db.createMilestone(trunk, branch, latestMilestone, config.MWM);
-
-      latestMilestoneHash = txs.get(0).getHash();
+      state.latestMilestoneTransactions = db.createMilestone(trunk, branch, state.latestMilestoneIndex, config.MWM);
 
-      if (config.broadcast) {
-        for (Transaction tx : txs) {
-          api.storeAndBroadcast(tx.toTrytes());
-        }
-        log.info("Broadcasted milestone.");
-      }
+      state.latestMilestoneHash = state.latestMilestoneTransactions.get(0).getHash();
 
-      log.info("Emitted milestone: " + latestMilestone);
+      // Do not store the state before broadcasting, since if broadcasting fails we should repeat the same milestone.
+      broadcastLatestMilestone();
 
       if (bootstrap >= 3) {
-        nextDepth = getNextDepth(depth, latestMilestoneTime);
+        nextDepth = getNextDepth(depth, state.latestMilestoneTime);
       } else {
         nextDepth = depth;
       }
 
       log.info("Depth: " + depth + " -> " + nextDepth);
 
       depth = nextDepth;
-      latestMilestoneTime = System.currentTimeMillis();
+      state.latestMilestoneTime = System.currentTimeMillis();
+
+      // Everything went fine, now we store
+      try {
+        storeState(state);
+      } catch (Exception e) {
+        String msg = "Error saving Compass state to file '" + statePath + "'!";
+
+        log.error(msg, e.getMessage());
```

Comment:
You can do one of the 3 options:
`log.error(msg, e)`  #entire stacktrace
`log.error("Error storing compass state file. {}", e.getMessage())` # {} is a placeholder for formatting the string. The path to the file will probably be printed by the error if it is an I/O error.
`log.error("msg" + e.getMessage())` #simple concatenation.

I think the current code will simply never print `e.getMessage()`